### PR TITLE
octopus: mgr/dashboard: Replace Protractor with Cypress 

### PR DESCRIPTION
**Pull request by tspmelo**
backport tracker: https://tracker.ceph.com/issues/45315
Also included:
- https://tracker.ceph.com/issues/45362
- https://tracker.ceph.com/issues/45408
- https://tracker.ceph.com/issues/44854
- https://tracker.ceph.com/issues/45351
- https://tracker.ceph.com/issues/45352

---

backport of https://github.com/ceph/ceph/pull/34285
parent tracker: https://tracker.ceph.com/issues/44812

this backport was staged using ceph-backport.sh version 15.1.1.389
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by votdev** [comment]
How shall we handle backports to Octopus including Cypress E2E tests when this PR is blocked? It does not make sense to change the E2E tests back to Protractor. If this is done, they will be lost or need to be adapted when this PR is merged.

**Comment by tspmelo** [comment]
I was just waiting to check if there were more PRs to include in this one, but we can merge this and do a normal backport for the future PRs.

File: src/pybind/mgr/dashboard/frontend/cypress/integration/page-helper.po.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+      return cy.contains('.datatable-body-cell-label', content);
+    } else {
+      return cy.get('.datatable-body-cell-label').first();
+    }
+  }
+
+  /**
+   * This was adapted for octopus, because row expansion doesn't exist in octopus.
+   * By changing the method code, we prevent making more changes in the actual
+   * test files.
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/cypress/integration/page-helper.po.ts:191]: Thanks a lot for this useful comment :)

**Comment by yuriw** [comment]
[wip-yuri7-testing-2020-05-19-2300-octopus](https://trello.com/c/EOVJPJeu/911-wip-yuri7-testing-2020-05-19-2300-octopus)

**Pull request merged**